_stbt/frameobject.py

```diff
@@ -197,12 +197,30 @@ def __bool__(self):
         """
         return bool(self.is_visible)
 
-    def __cmp__(self, other):
+    def __eq__(self, other):
         """
         Two instances of the same ``FrameObject`` type are considered equal if
         the values of all the public properties match, even if the underlying
         frame is different. All falsey FrameObjects of the same type are equal.
         """
+        return self.__cmp__(other) == 0
+
+    def __ne__(self, other):
+        return self.__cmp__(other) != 0
+
+    def __lt__(self, other):
+        return self.__cmp__(other) < 0
+
+    def __le__(self, other):
+        return self.__cmp__(other) <= 0
+
+    def __gt__(self, other):
+        return self.__cmp__(other) > 0
+
+    def __ge__(self, other):
+        return self.__cmp__(other) >= 0
+
+    def __cmp__(self, other):
```

Comment:
`functools.total_ordering` can help with this too.  Then all you need to define is `__eq__` and `__lt__` and it will define the rest.  This is fine though.

Example:

```
@functools.total_ordering
class Test(object):
    def __init__(self, a):
        self.a = a
    def __eq__(self, other):
        return self.a == other.a
    def __lt__(self, other):
        return self.a < other.a
```

Reply:
I think I didn't used `@functools.total_ordering` because I thought I was working on the MetaClass and I wasn't sure if the decorator would work in that case. Looking at it again now it turns out I was changing the actual `FrameObject` class. I'll leave it as is though.